lesson2/2_1.js

```diff
@@ -1,13 +1,13 @@
 // Реализовать функцию, которая принимает текст, 
 // и возвращает массив объектов со структурой { word: 'smth', length: 4, isCapitalized: false}
 
-var arrayBack = []
-function stringСonversion (vord) {
-    vord.split(' ').forEach(function callback(element,index){
-    arrayBack[index] = {'word': element, 'length': element.length, 'isCapitalised': element[0] == element[0].toUpperCase() }
-    })
-    return arrayBack
-}
+function stringСonversion (word) {
+ return word.replace(/[.,\/#!$%\^&\*;:{}=\_`~()]/g,"").split(' ').map(element => ([
+    element, 
+    element.length,
+    element[0] == element[0].toUpperCase()])).map(
+    item => new Object ({word: item[0], length: item[1], toUpperCase: item[2]}))
```

Comment:
Можно исключить промежуточный массив и сразу сериализовать объект:
```suggestion
 return word.replace(/[.,\/#!$%\^&\*;:{}=\_`~()]/g,"").split(' ').map(word => ({
    word, 
    length: word.length,
    isCapitalised: word[0] === word[0].toUpperCase()
}))
```

Reply:
Также, опционально можете добавить [поддержку разрыва строк](https://thinknetica.ponomarevlad.ru/2/1.html?url=https%3A%2F%2Fraw.githubusercontent.com%2FCaptain1965%2Fthink_js%2Ff8c89604093dc937402ef855b1f67e8c1a3a3f36%2Flesson2%2F2_1.js&func=stringСonversion#linebreak)